pkg/resource/deploy/plan_executor.go

```diff
@@ -83,6 +83,21 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 		}
 	}
 
+	// The set of -t targets provided on hte command line.  'nil' means 'update everything'.  Non-nill means 'update only
+	// in this set'
+	updateTargetsOpt, res := pe.plan.CheckTargets(opts.UpdateTargets)
+	if res != nil {
+		return res
+	}
+	destroyTargetsOpt, res := pe.plan.CheckTargets(opts.DestroyTargets)
+	if res != nil {
+		return res
+	}
+
+	if updateTargetsOpt != nil && destroyTargetsOpt != nil {
+		panic("Should not be possible to have both .DestroyTargets and .UpdateTargets")
```

Comment:
We normally use asserts instead of direct panics, right?

Reply:
```suggestion
		contract.Fail("Should not be possible to have both .DestroyTargets and .UpdateTargets")
```

Reply:
Done.